### DISCUSSION TITLE
Feature Request: New API Additions

### DISCUSSION THREAD

**Discussion by Altpapier** [HypixelDev/PublicAPI#542] [Ideas]
Hello 👋 

I have been using Hypixel's API for an extended amount of time now and wanted to suggest a few additions that are currently still missing and would be extremely useful! I have been implementing Hypixel's API into SkyHelper (the largest Skyblock Discord bot) for 2 years now and would be beneficial to use these features.

- **Museum**
It has been a long time (about half a year) since the museum was released and there is still no way to get the items a player has in his museum. It would also be beneficial to have for net worth.

- **Jacob Farming Contests**
Currently, many players obtain the Jacob contest data by manually collecting it from the calendar in Skyblock it would be cool if you could just get them from one endpoint.  

- **Placed Minions**
Would be nice to know players placed down minions as well as the items in their slots.

- **City Project Buffs**
It would also be nice to know the various rewards a player got from contributing to a city project like the -5% slayer cost buff.

- ~~**Crimson Essence**~~
~~There is every other essence in the API other than crimson.~~
(Added)

- ~~**Equipment** (Necklace, Belt, ...)~~
~~It would be beneficial to have for net worth.~~
(Added)

- ~~**Bestiary**~~
~~You can get the mob kills of a player in `stats` but bosses and some mobs cannot be accurately calculated as bosses only count as kills in the API if the player made the last hit and some mobs like voidling mobs, in the end, got added later to bestiary and didn't count the mobs killed before that.~~
(Added)

- ~~**Fishing Trophies**~~
~~Only just released but it would be cool if you could get the unique trophy fish catches.~~
(Added)

- ~~**Faction Reputation**~~
~~The nether update was just released, and it would be nice to know some player's progress on reputation~~
(Added)

- ~~**Bits** / **Fame Rank**
This has been released many years ago but unfortunately still is not in the API.~~
(Denied)

- ~~**Pelts**~~
~~Not that big of a deal but would still be cool to have.~~
(Added)

- ~~**Personal Bank**~~
~~Also, beneficial for net worth. This was a feature released way too long ago and get price reports for net worth every day because of players complaining their personal bank is not counting.~~
(Added)

- ~~**Social Skill**~~
~~This is in the API but only shows the experience obtained when the social skill was just released so before the social skill got reset for everyone. I already made a bug report for this probably a year ago now, but this unfortunately still has not changed.~~
(Added)

<ins>Suggestions from this discussion:

- ~~**Kuudra Statistics**~~
~~It would be nice to know how many runs a player has done of what tier and how many completions the player got and what the fastest time was and more...~~ (Suggested by @dukio)
(Added but fastest time is missing)

- **Playtime**
Just a nice thing to have. Can also be added to SkyHelper's /leaderboard command where the most addicted players can compare themselves. (Suggested by @Strqtz)

- **Kat**
What pet a player currently has sitting in kat.  (Suggested by @dukio)

- **Island Beacon**
What beacon tier the player has and how long that one is still active. (Suggested by @dukio)

- **Reaper Pepper**
Add the ability to see how many reaper peppers were already eaten by the player (Suggested by @dukio)

**Comment by MattTheCuber** [comment]
+1

**Comment by Mankifg** [comment]
I like that idea. , it shoud also  include chests on the island, but it will nead good system to not count same item twice

**Reply by Altpapier** [reply]
nah that would be too much data

**Reply by bishoperic** [reply]
I'm pretty sure that minions are stored in the island file, so unless they let us access the island file, I don't think we can get that.

**Comment by Strqtz** [comment]
Playtime would be cool as well.

**Reply by MattTheCuber** [reply]
Good point!

**Comment by dukio** [comment]
Kat too would be nice

**Comment by dukio** [comment]
Is this list only for stuff that could be useful for skyhelper or anything that isn't in API can make it to the list?

If the latter, Reaper Pepper count

**Reply by MattTheCuber** [reply]
It's for anything, what is reaper pepper?

**Reply by dukio** [reply]
A thing that you can eat up to 5 max and gives you +1hp each.
It's a permanent stat boost.

**Comment by Minikloon** [comment]
Monetization information such as bits and fame ranks will not be publicly disclosed.

**Reply by MattTheCuber** [reply]
That's fair.

**Comment by dukio** [comment]
Kuudra statistics. How many runs, what tier, how many completions, fastest time, etc...

**Comment by dukio** [comment]
Island data:
- Minions (list of placed minions with type, tier, upgrades, ...)
- Beacon tier and active perks
- Name, category

**Comment by Altpapier** [comment]
Added City Project Buffs to the list of suggestions

**Comment by MattTheCuber** [comment]
Thanks for all the recent additions Hypixel! This is amazing!

**Comment by Minikloon** [comment]
**New fields added today**
"crimson"
"equippment_contents"
"bestiary"
"trophy_fish"
"nether_island_player_data":
-> kuudra completions
-> dojo points and times
-> abiphone (list of strings)
-> matriarch, a few numbers
"trapper_quest"
"personal_bank_upgrade"
"social2" ("social" was not good enough!)
"pet_sitter
"reaper_peppers_eaten"
"soulflow"
"favorite_arrow"
"autopet"
"teleporter_pill_consumed"

**Comment by Minikloon** [comment]
I will lock this discussion, feel free to make to make a new one with pending suggestions.
